designer/client/src/Designer.tsx

```diff
@@ -31,16 +32,21 @@ function useData(props: Readonly<Props>): DataContextType {
    */
   const get = useCallback(async () => {
     const updated = await form.get(props.meta.id)
-    setData(updated)
-    return updated
+    const definition = updateConditions(updated)
+
+    setData(definition)
+
+    return definition
   }, [props])
 
   /**
    * Save form definition
    */
   const save = useCallback(
     async (updated: FormDefinition) => {
-      await form.save(props.meta.id, updateStartPage(updated))
+      const definition = updateConditions(updateStartPage(updated))
+      await form.save(props.meta.id, definition)
```

Comment:
@davidjamesstone If we're already fixing the data on `get()` do we need to fix it again on `save()`?

Reply:
Without it the saved state (DB) could drift/not be as up to date as it could be.

`get()` -> `updateConditions` -> `User updates a section title` -> `save()`

We need to call `updateConditions` before save else the changes to the section title will not be reflected in the condition display text (in the DB).

If we don't mind the drift, the fix up before `save` can be removed.